test/kaniko_task_test.go

```diff
@@ -175,19 +245,55 @@ func TestKanikoTaskRun(t *testing.T) {
 	}
 	podName := cluster.PodName
 	pods := c.KubeClient.Kube.CoreV1().Pods(namespace)
-	t.Logf("Retrieved pods for podname %s: %s\n", podName, pods)
+	logger.Infof("Retrieved pods for podname %s: %s\n", podName, pods)
 
-	req := pods.GetLogs(podName, &corev1.PodLogOptions{})
+	// get the logs for the kaniko step
+	req := pods.GetLogs(podName, &corev1.PodLogOptions{Container: "build-step-kaniko"})
 	readCloser, err := req.Stream()
 	if err != nil {
 		t.Fatalf("Failed to open stream to read: %v", err)
 	}
-	defer readCloser.Close()
-	var buf bytes.Buffer
-	out := bufio.NewWriter(&buf)
-	_, err = io.Copy(out, readCloser)
-	if !strings.Contains(buf.String(), kanikoBuildOutput) {
-		t.Fatalf("Expected output %s from pod %s but got %s", kanikoBuildOutput, podName, buf.String())
+	bs, err := ioutil.ReadAll(readCloser)
+	if err != nil {
+		t.Fatalf("Failed to read pod log stream: %v", err)
+	}
+	initLogs := string(bs)
+	readCloser.Close()
+
+	// get the pods' logs
+	req = pods.GetLogs(podName, &corev1.PodLogOptions{})
+	readCloser, err = req.Stream()
+	if err != nil {
+		t.Fatalf("Failed to open stream to read: %v", err)
+	}
+	readCloser.Close()
+	bs, err = ioutil.ReadAll(readCloser)
+	if err != nil {
+		t.Fatalf("Failed to read pod log stream: %v", err)
 	}
+	podLogs := string(bs)
 
+	// check the logs match what we expect
+	if !strings.Contains(podLogs, kanikoBuildOutput) {
+		t.Fatalf("Expected output %s from pod %s but got %s", kanikoBuildOutput, podName, string(bs))
+	}
+	// make sure the pushed digest matches the one we pushed
+	re := regexp.MustCompile("digest: (sha256:\\w+)")
+	ms := re.FindAllStringSubmatch(initLogs, -1)
+	if len(ms) != 1 && len(ms[0]) < 2 {
+		t.Fatalf("Expected to find one image digest in the build output")
+	}
+	localDigest := ms[0][1]
+	// pull digest from pushed image
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		t.Errorf("Expected to be able to pull remote ref %s: %v", ref, err)
+	}
+	digest, err := img.Digest()
+	if err != nil {
+		t.Errorf("Expected to get digest for image %s: %v", img, err)
+	}
+	if digest.String() != localDigest {
+		t.Errorf("Expected built digest %s to match remote digest %s", localDigest, digest)
+	}
```

Comment:
what do you think about breaking some of this out into separate functions? i think there's a pretty good chance we might want to use this logic more than once, and it would be easier to read what the test is doing